### ISSUE TITLE
[Notice] Undefined index: children 

### ISSUE THREAD

**Issue by nyeholt** [silbinarywolf/silverstripe-components#38]
For a template structure like

```
<section class="Section">
    <:Banner>
        <:SearchDashlet Form="$SearchDashletForm"></:SearchDashlet>
    </:Banner>
</section>
```

I get the following error. So far as I can see, the template parser isn't correctly handling nested component templates?

```
53 $_props = array();
54 $_props['children'] = '';
55 $_props['children'] .= '
56         ';
57 
58 $_props = array();
59 $_props['Form'] = array();
60 $_props['Form'][] = $scope->locally()->obj('SearchDashletForm', null, true)->self();
61 $_props['Form'] = SilverStripe\Core\Injector\Injector::inst()->get('Symbiote\Components\ComponentService')->createProperty('Form', $_props['Form']);
62 $_props['children'] .= SilverStripe\Core\Injector\Injector::inst()->get('Symbiote\Components\ComponentService')->renderComponent('SearchDashlet', $_props, $scope);
63 unset($_props);
```

**Comment by JoshuaCarter** [comment]
Issue likely here: https://github.com/symbiote/silverstripe-components/blob/bf87fb793d98eaff22ab58a6ca61ebac8c8dd462/src/Symbiote/Components/ComponentService.php#L211

**Comment by JoshuaCarter** [comment]
If I insert:
```
$php = str_replace("\$_props = array();\n", "", $php);
$php = str_replace("unset(\$_props);\n", "", $php);
```
Above this line:
https://github.com/symbiote/silverstripe-components/blob/bf87fb793d98eaff22ab58a6ca61ebac8c8dd462/src/Symbiote/Components/ComponentService.php#L175

The generated code for children is corrected and the errors go away.

**Comment by JoshuaCarter** [comment]
@nyeholt PHP template code for 'children' needed to have _props assignment/unset stripped so that it doesn't overwrite/unset the parent's data.

PR here: https://github.com/symbiote/silverstripe-components/pull/39

Note that this *does* fix nested components, but the template you're using doesn't render them due to the `<% if $Title || $Content %>` check failing in Banner.ss. If I move `$children` above that, it works.

**Comment by nyeholt** [comment]
Closed via #39
